### PR TITLE
[java] Add Untrusted Deserialization tests

### PR DESCRIPTION
**Pull request by Mariovido**
## Motivation

Add system test for a new vulnerability (Untrusted Deserialization)

## Changes

Added test for the Untrusted Deserialization vulnerability.
Added only system test for weblog variant **springboot**.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/_features.py [hunk 1]
```diff
@@ -2266,3 +2266,13 @@ def container_guardrail(test_object):
...
         return test_object
+
+    @staticmethod
+    def iast_sink_untrusted_deserialization(test_object):
+        """
+        IAST Sink: Untrusted Deserialization
+
+        https://feature-parity.us1.prod.dog/#/?feature=311
+        """
+        pytest.mark.features(feature_id=311)(test_object)
```
Comment (jandro996) [inline utils/_features.py:2277]: We need to add it to feature parity, this id is provisional

Comment (Mariovido) [reply utils/_features.py]: Updated to the new feature_id

**Review by ValentinZakharov — APPROVED** [review]

**Review by jandro996 — APPROVED** [review]

**Comment by cbeauchesne** [comment]
Could you rebase your branch? it should fix some CI errors

**Comment by Mariovido** [comment]
> Could you rebase your branch? it should fix some CI errors

@cbeauchesne Already rebased

**Review by pawelchcki — APPROVED** [review]

**Pull request merged**